provisioner/terraform/executor.go

```diff
@@ -259,6 +286,53 @@ func getStateFilePath(workdir string) string {
 	return filepath.Join(workdir, "terraform.tfstate")
 }
 
+func getTerraformLockFilePath(workdir string) string {
+	return filepath.Join(workdir, ".terraform.lock.hcl")
+}
+
+// generateFileDiff generates a simple diff between two file contents.
+// Returns empty string if files are identical.
+func generateFileDiff(beforeContent, afterContent []byte) string {
+	if bytes.Equal(beforeContent, afterContent) {
+		return ""
+	}
+
+	// Simple line-by-line diff
+	beforeLines := strings.Split(string(beforeContent), "\n")
+	afterLines := strings.Split(string(afterContent), "\n")
+
+	var diff strings.Builder
+	diff.WriteString("--- .terraform.lock.hcl (before terraform init)\n")
+	diff.WriteString("+++ .terraform.lock.hcl (after terraform init)\n")
+
+	// Simple diff showing added/removed lines
+	beforeMap := make(map[string]bool)
+	for _, line := range beforeLines {
+		beforeMap[line] = true
+	}
+
+	afterMap := make(map[string]bool)
+	for _, line := range afterLines {
+		afterMap[line] = true
+	}
+
+	// Show removed lines
+	for _, line := range beforeLines {
+		if !afterMap[line] && strings.TrimSpace(line) != "" {
+			diff.WriteString("- " + line + "\n")
+		}
+	}
+
+	// Show added lines
+	for _, line := range afterLines {
+		if !beforeMap[line] && strings.TrimSpace(line) != "" {
+			diff.WriteString("+ " + line + "\n")
+		}
+	}
+
+	return diff.String()
+}
```

Comment:
See https://pkg.go.dev/github.com/google/go-cmp/cmp#Diff
We already use this and its output is very readable!